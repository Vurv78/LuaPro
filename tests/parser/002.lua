function a.b:c(x, ...) end