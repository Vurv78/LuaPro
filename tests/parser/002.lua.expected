local lupa = require("src.lib.mod")
local Node, Token = lupa.Node, lupa.Token

return Node.new(Node.Variant.Chunk, {
	Node.new(Node.Variant.Function,{
		false,
		{
				{ false, Token.new(Token.Variant.Identifier, "a") },
				{ false, Token.new(Token.Variant.Identifier, "b") },
				{ true, Token.new(Token.Variant.Identifier, "c") }
			},
		{
				Token.new(Token.Variant.Identifier, "x"),
				Token.new(Token.Variant.Vararg, nil)
			},
		Node.new(Node.Variant.Chunk, {})
	})
})